httpx/_auth.py

```diff
@@ -228,6 +228,7 @@ def _parse_challenge(
         `Digest realm="realm@host.com",qop="auth,auth-int",nonce="abc",opaque="xyz"`
         """
         scheme, _, fields = auth_header.partition(" ")
+        entity = response.read()
```

Comment:
Actually we need to use `requires_response_body = True` on the auth class, and access `response.content` here.

This will ensure that the authentication class works with both sync `response.read()` and async `await response.aread()` cases.

There's some documentation on that here...

https://www.python-httpx.org/advanced/#customizing-authentication